lang/java/reef-io/src/test/java/org/apache/reef/services/network/NetworkConnectionServiceTest.java

```
@@ -188,33 +183,35 @@ public void testMessagingNetworkConnServiceRate() throws Exception {
       final int numMessages = 300000 / (Math.max(1, size / 512));
       final Monitor monitor = new Monitor();
       final Codec<String> codec = new StringCodec();
-      final NetworkMessagingTest messagingTest = new NetworkMessagingTest(localAddress);
-      messagingTest.registerTestConnectionFactory(groupCommClientId, numMessages, monitor, codec);
-      final Connection<String> conn = messagingTest.getConnectionFromSenderToReceiver(groupCommClientId);
-
-      // build the message
-      final StringBuilder msb = new StringBuilder();
-      for (int i = 0; i < size; i++) {
-        msb.append("1");
-      }
-      final String message = msb.toString();
-
-      long start = System.currentTimeMillis();
-      try {
-        conn.open();
-        for (int count = 0; count < numMessages; ++count) {
-          // send messages to the receiver.
-          conn.write(message);
+      try (final NetworkMessagingTestService messagingTestService = new NetworkMessagingTestService(localAddress)) {
+        messagingTestService.registerTestConnectionFactory(groupCommClientId, numMessages, monitor, codec);
+
+        // build the message
+        final StringBuilder msb = new StringBuilder();
+        for (int i = 0; i < size; i++) {
+          msb.append("1");
         }
-        monitor.mwait();
-      } catch (NetworkException e) {
-        e.printStackTrace();
-      }
-      final long end = System.currentTimeMillis();
+        final String message = msb.toString();
+
+        try (final Connection<String> conn = messagingTestService.getConnectionFromSenderToReceiver(groupCommClientId)) {
+
+          long start = System.currentTimeMillis();
+          try {
+            conn.open();
+            for (int count = 0; count < numMessages; ++count) {
+              // send messages to the receiver.
+              conn.write(message);
+            }
+            monitor.mwait();
+          } catch (NetworkException e) {
+            e.printStackTrace();
```

Same as above: Is it safe to swallow the exception? If so, please document why.